proposed/config-meta.md

```diff
@@ -0,0 +1,134 @@
+The goal set by the Config PSR is to standardize how factories uses a configuration to create instances, support for 
+auto discovery of needed configuration, to reduce boilerplate code and to make it more readable and easier to understand. 
+It can also be used to build the name of the *Container* entry.
+
+### 3.2 Non-goals
+* It's not a goal to define a whole project/library configuration structure.
+* It's not a goal to validate the configuration option values.
+
+## 4. History
+Before submitting the Config PSR to the PHP-FIG, the interfaces was first proposed in a project named 
+[interop-config](https://github.com/sandrokeil/interop-config).
+
+The goal of the project was to provide a test-bed for implementing the interfaces and implementation and to pave the way 
+for the Config PSR.
+
+## 5. Interfaces
+There are six interfaces and some interfaces extends from another interface to ensure the uniform configuration structure. 
+If you confused about the naming of the interfaces or methods, note that I have used the Domain Driven Design approach.
+
+### 5.1 RequiresConfig
+This is the main interface and describes the default configuration structure. Since we have Composer more and more 
+libraries can be combined to create individual projects. So it is only logical to start the configuration structure 
+with a `vendorName`. A vendor has multiple packages so the next structure depth is `packageName`.
+
+As an array it looks like this:
+
+```php
+[
+    // vendor name
+    'doctrine' => [
+        // package name
+        'connection' => [
+        ],
+    ],
+];
+```
+
+The factory retrieves the configuration options e.g. from the *Container*. The method `options($config)` extracts the 
+configuration options from the given array or an object which implements `ArrayAccess` depending on the implemented 
+interfaces and optionally does a mandatory option check. If `ProvidesDefaultOptions` interface is defined, the default 
+options are merged and overridden of the provided config.
+
+
+The `options` method:
+
+* returns the options depending on the implemented interfaces, optionally mandatory option check and merging of default options
+* throws an `InvalidArgumentException` if the parameter `config` has the wrong type
+* throws an `OutOfBoundsException` If the vendor name was not found in configuration
+* throws an `OptionNotFoundException` if no options are found
+* throws a `MandatoryOptionNotFoundException` if a mandatory option is missing
+
+
+The `canRetrieveOptions` method:
+
+* checks if options are available depending on implemented interfaces and checks that the retrieved options are an array 
+or have implemented \ArrayAccess. The `RequiresContainerId` interface is optional but MUST be supported.
+
+### 5.2 RequiresContainerId
+Factories allow to create more than one instance of the same class with another configuration. This can be achieved with 
+the third structure depth `containerId`. Now the structure is fully expanded.
```

Comment:
This is possible, but it makes the config more unreadable and makes no sense. I think the uniform structure is missing in this case, beause you are mixing the package name with an id or something else. It's up to you to use this interface. There is no check, if the vendor/package name matches the current factory. Only the configuration array is checked for vendor/package.

The `RequriesContainerId` interface is only needed for factories which creates multiple instances with different configuration e.g. database connections (read/write). There are many PHP libraries which use this approach.

* [DoctrineORMModule](https://github.com/doctrine/DoctrineORMModule/blob/master/docs/configuration.md)
* [Prooph Service Bus](https://github.com/prooph/service-bus/blob/develop/config/prooph_service_bus.config.php)
* Zend Framework 2 and other frameworks in some cases